packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -78,4 +94,103 @@ export class ApplicationExtension<
         const {allRoutes} = params
         return allRoutes
     }
+
+    /**
+     * Called on the server to serialize the extension's asynchronous routes that will be sent to the client.
+     *
+     * @returns SerializedRouteProps - The serialized extension data.
+     * @returns null if there's no asynchronous state that needs to be serialized
+     * @throws Error if the routes have not been loaded.
+     */
+    public serializeAsyncRoutes(): SerializedRouteProps[] | null {
+        if (typeof this.getRoutesAsync === 'undefined') return null
+
+        if (this._cachedRoutes === null) {
+            throw new Error(`Routes have not been loaded. Call getRoutesAsync() before serializing`)
+        }
```

Comment:
Maybe you touch on this a little bit here with regards to my above comments. I'm still wondering if we need to have 3 states, throw, null and valid value. Seems like it might be more difficult to understand.

Reply:
I've returned an empty array whenever possible however I feel we still need to have the null case for the initial cachedRoutes value.

The reason is the `cacheMethodResults` only caches the value when the property is null or undefined. And since this is a general function I don't think caching when the property is an empty array would be a good idea. But open to suggestions.

Here are the changes: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2300/commits/3d5febc0286428d786b46db514ed06d907aa8119

Reply:
use private variable for cachedRoutesInvoked!